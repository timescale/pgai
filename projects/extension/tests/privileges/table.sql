\set users {bob,fred,alice,jill}

-- check table privileges
\! rm -f schema.actual
select
  n.nspname as "schema"
, k.relname as "table"
, u as "user"
, p as "privilege"
, case has_table_privilege(u, k.oid, p) when true then 'YES' else 'no' end as granted
from unnest(:'users'::text[]) u
inner join pg_namespace n on (n.nspname = any(array['ai', 'wiki']))
inner join pg_class k on (n.oid = k.relnamespace and k.relkind in ('r', 'p'))
cross join unnest(array['select', 'insert', 'update', 'delete']) p
order by n.nspname, k.relname, u, p
\g (format=aligned) table.actual
