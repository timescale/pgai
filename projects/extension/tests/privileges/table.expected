 schema |        table         | user  | privilege | granted 
--------+----------------------+-------+-----------+---------
 ai     | _secret_permissions  | alice | delete    | YES
 ai     | _secret_permissions  | alice | insert    | YES
 ai     | _secret_permissions  | alice | select    | YES
 ai     | _secret_permissions  | alice | update    | YES
 ai     | _secret_permissions  | bob   | delete    | no
 ai     | _secret_permissions  | bob   | insert    | no
 ai     | _secret_permissions  | bob   | select    | no
 ai     | _secret_permissions  | bob   | update    | no
 ai     | _secret_permissions  | fred  | delete    | no
 ai     | _secret_permissions  | fred  | insert    | no
 ai     | _secret_permissions  | fred  | select    | no
 ai     | _secret_permissions  | fred  | update    | no
 ai     | _secret_permissions  | jill  | delete    | no
 ai     | _secret_permissions  | jill  | insert    | no
 ai     | _secret_permissions  | jill  | select    | no
 ai     | _secret_permissions  | jill  | update    | no
 ai     | _vectorizer_q_1      | alice | delete    | YES
 ai     | _vectorizer_q_1      | alice | insert    | YES
 ai     | _vectorizer_q_1      | alice | select    | YES
 ai     | _vectorizer_q_1      | alice | update    | YES
 ai     | _vectorizer_q_1      | bob   | delete    | no
 ai     | _vectorizer_q_1      | bob   | insert    | no
 ai     | _vectorizer_q_1      | bob   | select    | no
 ai     | _vectorizer_q_1      | bob   | update    | no
 ai     | _vectorizer_q_1      | fred  | delete    | YES
 ai     | _vectorizer_q_1      | fred  | insert    | YES
 ai     | _vectorizer_q_1      | fred  | select    | YES
 ai     | _vectorizer_q_1      | fred  | update    | YES
 ai     | _vectorizer_q_1      | jill  | delete    | YES
 ai     | _vectorizer_q_1      | jill  | insert    | YES
 ai     | _vectorizer_q_1      | jill  | select    | YES
 ai     | _vectorizer_q_1      | jill  | update    | YES
 ai     | migration            | alice | delete    | YES
 ai     | migration            | alice | insert    | YES
 ai     | migration            | alice | select    | YES
 ai     | migration            | alice | update    | YES
 ai     | migration            | bob   | delete    | no
 ai     | migration            | bob   | insert    | no
 ai     | migration            | bob   | select    | no
 ai     | migration            | bob   | update    | no
 ai     | migration            | fred  | delete    | no
 ai     | migration            | fred  | insert    | no
 ai     | migration            | fred  | select    | no
 ai     | migration            | fred  | update    | no
 ai     | migration            | jill  | delete    | no
 ai     | migration            | jill  | insert    | no
 ai     | migration            | jill  | select    | no
 ai     | migration            | jill  | update    | no
 ai     | vectorizer           | alice | delete    | YES
 ai     | vectorizer           | alice | insert    | YES
 ai     | vectorizer           | alice | select    | YES
 ai     | vectorizer           | alice | update    | YES
 ai     | vectorizer           | bob   | delete    | no
 ai     | vectorizer           | bob   | insert    | no
 ai     | vectorizer           | bob   | select    | no
 ai     | vectorizer           | bob   | update    | no
 ai     | vectorizer           | fred  | delete    | no
 ai     | vectorizer           | fred  | insert    | no
 ai     | vectorizer           | fred  | select    | YES
 ai     | vectorizer           | fred  | update    | no
 ai     | vectorizer           | jill  | delete    | YES
 ai     | vectorizer           | jill  | insert    | YES
 ai     | vectorizer           | jill  | select    | YES
 ai     | vectorizer           | jill  | update    | YES
 ai     | vectorizer_errors    | alice | delete    | YES
 ai     | vectorizer_errors    | alice | insert    | YES
 ai     | vectorizer_errors    | alice | select    | YES
 ai     | vectorizer_errors    | alice | update    | YES
 ai     | vectorizer_errors    | bob   | delete    | no
 ai     | vectorizer_errors    | bob   | insert    | no
 ai     | vectorizer_errors    | bob   | select    | no
 ai     | vectorizer_errors    | bob   | update    | no
 ai     | vectorizer_errors    | fred  | delete    | no
 ai     | vectorizer_errors    | fred  | insert    | no
 ai     | vectorizer_errors    | fred  | select    | no
 ai     | vectorizer_errors    | fred  | update    | no
 ai     | vectorizer_errors    | jill  | delete    | YES
 ai     | vectorizer_errors    | jill  | insert    | YES
 ai     | vectorizer_errors    | jill  | select    | YES
 ai     | vectorizer_errors    | jill  | update    | YES
 wiki   | post                 | alice | delete    | YES
 wiki   | post                 | alice | insert    | YES
 wiki   | post                 | alice | select    | YES
 wiki   | post                 | alice | update    | YES
 wiki   | post                 | bob   | delete    | no
 wiki   | post                 | bob   | insert    | no
 wiki   | post                 | bob   | select    | no
 wiki   | post                 | bob   | update    | no
 wiki   | post                 | fred  | delete    | no
 wiki   | post                 | fred  | insert    | no
 wiki   | post                 | fred  | select    | YES
 wiki   | post                 | fred  | update    | no
 wiki   | post                 | jill  | delete    | no
 wiki   | post                 | jill  | insert    | no
 wiki   | post                 | jill  | select    | YES
 wiki   | post                 | jill  | update    | no
 wiki   | post_embedding_store | alice | delete    | YES
 wiki   | post_embedding_store | alice | insert    | YES
 wiki   | post_embedding_store | alice | select    | YES
 wiki   | post_embedding_store | alice | update    | YES
 wiki   | post_embedding_store | bob   | delete    | no
 wiki   | post_embedding_store | bob   | insert    | no
 wiki   | post_embedding_store | bob   | select    | no
 wiki   | post_embedding_store | bob   | update    | no
 wiki   | post_embedding_store | fred  | delete    | no
 wiki   | post_embedding_store | fred  | insert    | YES
 wiki   | post_embedding_store | fred  | select    | YES
 wiki   | post_embedding_store | fred  | update    | YES
 wiki   | post_embedding_store | jill  | delete    | no
 wiki   | post_embedding_store | jill  | insert    | YES
 wiki   | post_embedding_store | jill  | select    | YES
 wiki   | post_embedding_store | jill  | update    | YES
(112 rows)

